processes/record_ingestor.py

```diff
@@ -0,0 +1,113 @@
+import json
+import multiprocessing
+import os
+from sqlalchemy.orm import Session
+
+from logger import create_log
+from managers import DBManager, RabbitMQManager, S3Manager
+from model import Record, FRBRStatus
+from services.sources.source_service import SourceService
+from . import utils
+
+logger = create_log(__name__)
+
+
+class RecordIngestor:
+
+    def __init__(self, source_service: SourceService, source: str, *args):
+        self.params = utils.parse_process_args(*args)
+        self.source = source
+        self.source_service = source_service
+
+    def ingest(self) -> int:
+        ingest_count = 0
+
+        try:
+            records = self.source_service.get_records(
+                start_timestamp=utils.get_start_datetime(process_type=self.params.process_type, ingest_period=self.params.ingest_period),
+                offset=self.params.offset,
+                limit=self.params.limit
+            )
+
+            ingest_pipeline_pool = multiprocessing.Pool(processes=4)
+            record_ingest_processes = [ingest_pipeline_pool.apply_async(self._ingest_record, args=(record,)) for record in records]
```

Comment:
What are the 4 processes that will be in the multiprocessing pool?

Reply:
Basically there are 4 subprocesses that will ingest records at a given time. 

This matches what the S3 process does - just spins up 4 processes.